src/main/java/io/gravitee/policy/basicauth/BasicAuthenticationPolicy.java

```diff
@@ -53,12 +69,29 @@ public BasicAuthenticationPolicy(BasicAuthenticationPolicyConfiguration basicAut
     public void onRequest(Request request, Response response, ExecutionContext executionContext, PolicyChain policyChain) {
         String authorizationHeader = request.headers().getFirst(HttpHeaderNames.AUTHORIZATION);
 
-        if (authorizationHeader == null || authorizationHeader.trim().isEmpty()) {
-            sendAuthenticationFailure(response, policyChain);
-            return;
-        }
+        if (
+            authorizationHeader == null ||
+            authorizationHeader.trim().isEmpty() ||
+            !authorizationHeader.toUpperCase().startsWith(BASIC_AUTHENTICATION_VALUE)
+        ) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(
+                    ERROR_MESSAGE_FORMAT,
+                    executionContext.getAttribute(ATTR_API),
+                    request.id(),
+                    request.path(),
+                    INVALID_AUTH_HEADER_ERROR_MESSAGE
+                );
+            } else if (LOGGER.isWarnEnabled()) {
+                LOGGER.warn(
+                    ERROR_MESSAGE_FORMAT,
+                    executionContext.getAttribute(ATTR_API),
+                    request.id(),
+                    request.path(),
+                    INVALID_AUTH_HEADER_ERROR_MESSAGE
+                );
+            }
```

Comment:
To avoid this big piece of code, it could be moved to a private method that deals with the debug / warn and take the error message in parameter such as:

```java
private void logErrorMessage(String errorMessage, ExecutionContext ctx) {
     if (LOGGER.isDebugEnabled()) {
                LOGGER.debug(
                    ERROR_MESSAGE_FORMAT,
                    ctx.getAttribute(ATTR_API),
                    ctx.request().id(),
                    ctx.request().path(),
                    errorMessage
                );
            } else if (LOGGER.isWarnEnabled()) {
                LOGGER.warn(
                    ERROR_MESSAGE_FORMAT,
                    ctx.getAttribute(ATTR_API),
                    ctx.request().id(),
                    ctx.request().path(),
                    errorMessage
                );
            }
}
```

wdyt?

Reply:
Why do we need to handle both log level ? Can't we just use the "WARN" level for this log ?